src/libraries/System.Linq/src/System/Linq/OrderedEnumerable.cs

```diff
@@ -158,6 +173,16 @@ internal override CachingComparer<TElement> GetComparer(CachingComparer<TElement
                 : new CachingComparerWithChild<TElement, TKey>(_keySelector, _comparer, _descending, childComparer);
             return _parent != null ? _parent.GetComparer(cmp) : cmp;
         }
+
+        protected override bool CanInPlaceSort()
```

Comment:
This would be a good place to write a comment detailing the motivations for this optimization.